app/controllers/staff/home_controller.rb

```diff
@@ -5,8 +5,24 @@ def show
     if current_user.service_owner?
       @delivery_partner_organisations = Organisation.delivery_partners
       authorize @delivery_partner_organisations
+      render :service_owner
     else
-      redirect_to organisation_path(current_user.organisation)
+      @grouped_programmes = fetch_grouped_programmes_for(current_user.organisation, :current)
+      render :delivery_partner
     end
   end
+
+  private def fetch_grouped_programmes_for(organisation, scope)
+    activities = policy_scope(
+      Activity.includes(
+        :organisation,
+        parent: [:parent, :organisation],
+        child_activities: [:child_activities, :organisation, :parent]
+      ).programme
+       .send(scope)
+    )
+    activities = activities.where(extending_organisation: organisation)
+    activities.order(:roda_identifier_compound)
+      .group_by(&:parent)
+  end
```

Comment:
It looks like there's almost identical code being used in the ActivitiesController. Can we break this out into a service or similar?

Reply:
Yeah, I plan to, just as other work. I thought I covered this in a comment? Anyway - it is also not quite working how I think it probably should, but I want to keep to the acceptance on this card and make this better and probably add it to Activity.

Are you okay to let this in if I make sure the card is written up?

Reply:
Oop! I have already: https://trello.com/c/II0qSP5w

Reply:
Ah, so you did - sorry. Missed that! Yep - go for it!